### PR TITLE
revamp extend classes

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: core/domain/issue/models/issue-model.ts [hunk 1]
```diff
@@ -11,12 +11,12 @@ interface GithubIssueResponse extends _GithubIssueResponse {
...
 }
 
-export interface IssueInterface extends ListIssueInterface, GithubIssueResponse {}
-
-const ExtendedWithListIssue = extendMultipleClasses(mapApiToClass<GithubIssueResponse>(), ListIssue);
-
-export class Issue extends ExtendedWithListIssue implements IssueInterface {
-  constructor(protected readonly props: GithubIssueResponse) {
+export interface IssueInterface extends GithubIssueResponse {}
+export class Issue extends mapApiToClass<GithubIssueResponse>() implements IssueInterface {
```
Comment (haydencleary) [inline core/domain/issue/models/issue-model.ts:15]: `mapApiToClass` doesn't go inside `extendClasses` ?

File: core/domain/issue/models/issue-model.ts [hunk 2]
```diff
@@ -11,12 +11,12 @@ interface GithubIssueResponse extends _GithubIssueResponse {
...
+  constructor(protected props: GithubIssueResponse) {
     super(props);
+    Object.assign(this, props);
```
Comment (haydencleary) [inline core/domain/issue/models/issue-model.ts:18]: This is still required here ?

File: core/domain/issue/models/issue-model.ts (hunk 1 above)
Comment (alexbeno) [reply core/domain/issue/models/issue-model.ts:15]: If we want to get autocomplete of the props we should include it inside the extends

File: core/domain/issue/models/issue-model.ts (hunk 2 above)
Comment (alexbeno) [reply core/domain/issue/models/issue-model.ts:18]: let's remove this

**Pull request merged**